src/java/org/apache/cassandra/metrics/TableMetrics.java

```diff
@@ -1190,32 +1134,59 @@ private LatencyMetrics createLatencyMetrics(String namePrefix, LatencyMetrics ..
      * @return true if first time metric with that name has been registered
      */
     private boolean register(String name, String alias, Metric metric)
+    {
+        return register(name, alias, null, metric);
+    }
+
+    /**
+     * Registers a metric to be removed when unloading CF.
+     * 
+     * @param name the name of the metric registered with the "Table" type
+     * @param alias the name of the metric registered with the legacy "ColumnFamily" type
+     * @param deprecated an optionally null deprecated name for the metric registered with the "Table"
+     * 
+     * @return true if first time metric with that name has been registered
+     */
+    private boolean register(String name, String alias, String deprecated, Metric metric)
     {
         boolean ret = allTableMetrics.putIfAbsent(name, ConcurrentHashMap.newKeySet()) == null;
         allTableMetrics.get(name).add(metric);
-        all.add(() -> releaseMetric(name, alias));
+        all.add(() -> releaseMetric(name, alias, deprecated));
         return ret;
     }
 
-    private void releaseMetric(String metricName, String metricAlias)
+    private void releaseMetric(String tableMetricName, String cfMetricName, String tableMetricAlias)
```

Comment:
`tableMetricAlias` threw me off as it is the deprecated version, which should exist in Table and CF metrics